oeps/oep-0044-bp-changelogs.rst

```diff
@@ -0,0 +1,294 @@
+
+.. _semantic-release: https://github.com/semantic-release/semantic-release
+.. _GitHub Release changelog for edx/frontend-build: https://github.com/edx/frontend-build/releases
+
+Python Libraries
+~~~~~~~~~~~~~~~~
+
+Like all Open edX repositories, Python libraries should also include a CHANGELOG.rst. This file may also simply point to the GitHub Releases. Here is an example CHANGELOG.rst::
+
+  Changelog
+  =========
+
+  All notable changes to this project will be documented as `GitHub Releases`_.
+
+  This project adheres to `Semantic Versioning`_.
+
+  .. _GitHub Releases: https://github.com/edx/frontend-build/releases
+  .. _Semantic Versioning: https://semver.org/spec/v2.0.0.html
+
+Many Python libraries publish packages to PyPI when a version is tagged. Publishing a GitHub Release both tags the version and allows one to write an appropriate changelog message. This is the recommended process at this time.
+
+Every published release should have a corresponding changelog entry.
+
+Application and Non-Library Repositories
+----------------------------------------
+
+When it comes to changelogs, the main difference between application and non-library repositories, and the eariler recommendations for library repositories, is that non-library code is not versioned.
+
+Since this code has no versions, the recommendations differ in the following ways:
+
+* The changelog entries will live in the CHANGELOG.rst file, following the recommendations detailed in `Changelogs and Changelog Entries`_.
+* Changelog entries should start with a date, rather than a date and version, because these repositories are not versioned.
+
+Changelog for edx-platform
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The edx-platform repository should have a CHANGELOG.rst, just like all other repositories.
+
+* Many notes that could potentially live in the edx-platform CHANGELOG.rst get added to the `Open Edx Release Planning`_ pages which have historically been created for each named release. Worst case, edx-platform notes are manually added to the CHANGELOG.rst instead, and at least the documentation stays more closely tied to the code.
+
+* Entries could be grouped by day if they could be added at a faster cadence (see `Tooling and Automation`_). Otherwise, date ranges should be used.
+
+* Attempting to add changelog entries at a very faster cadence without automation is likely to cause too many merge conflict issues. However, this should not stop people from adding entries immediately so we can learn.
+
+* As with all changelogs, it is acceptable and appropriate to clean up or enhance or add changelog entries as necessary.
+
+* More work is clearly required to determine how to iterate and improve the process for a CHANGELOG.rst in edx-platform.
+
+* The CHANGELOG.rst will also likely need to be broken into multiple files, but that can be a local decision.
```

Comment:
Why would it need to be broken into multiple files? Also, "local" decision? This is in the edx-platform section.

Reply:
I pulled this section so it really could live in an ADR when and if it is actually necessary.  Maybe it never will be.  This was about whether the changelog gets too long and we want to break it by year or anything else.